### PR TITLE
feat(table): added frozen header, frozen column, permanent scrollbars

### LINKED ISSUE #2457: table: iteration 3 module rollout
### Description

We've previously completed two iterations of updates rolling out the `table` module. For this iteration 3, add the following features:

1. Add permanent horizontal scrollbars for available offscreen content affordance.
2. Add table container height modifier for vertical scrolling with an overflow.
3. Add fixed header, which is related to 2, but may have some nuance. To be clarified.

Additionally, if the specifications are decided in time.

1. Add table data loading indicator for table state during filtering, sorting, pagination change, etc.

### Notes

Main issue for reference: https://github.com/eBay/skin/issues/1900

### Engineering pre-requisites

- [X] Supports dark mode.
- [X] Supports responsive design.
- [X] Tokens have been provided (or already exist).

### PR DESCRIPTION
**Pull request by ArtBlue**
Fixes #2457

- [X] This PR contains CSS changes
- [ ] This PR does not contain CSS changes

## Description
Added the next set of features to `table`:

* added frozen header
* added full height table modifier
* added frozen column
* added full height table override
* added permanent scrollbars

## Notes
For the permanent scrollbars, I created a utility mixin to be used by a public utility class as well as intrinsically for `table` without the use of the utility class.

## Screenshots

<kbd>image</kbd>

<kbd>image</kbd>

<kbd>image</kbd>

## Checklist

- [X] I verify the build is in a non-broken state
- [X] I verify all changes are within scope of the linked issue

- [X] I regenerated all CSS files under dist folder
- [X] I tested the UI in all supported browsers
- [x] I did a visual regression check of the components impacted by doing a Percy build and approved the build
- [X] I tested the UI in dark mode and RTL mode
- [X] I added/updated/removed Storybook coverage as appropriate

### CODE REVIEW

File: src/sass/table/table.scss [hunk 1]
```diff
@@ -64,6 +68,34 @@ $density-relaxed-img-radius: 8px;
     padding: $density-relaxed-padding;
 }
 
+.table--frozen-header thead {
```
Comment (bill-min) [inline src/sass/table/table.scss:71]: I might not have full context, but I am see everywhere else using `sticky` instead of `frozen` even in figma playbook.

File: src/sass/table/table.scss [hunk 2]
```diff
@@ -131,6 +163,11 @@ $density-relaxed-img-radius: 8px;
     text-align: center;
     width: 48px;
 }
+/* Not sure of downstream dev consumers' ability to utilize this.
+   Convert to long-hand if problematic. */
+.table--mode-selection tr:has(input:checked) :where(td, th) {
```
Comment (bill-min) [inline src/sass/table/table.scss:168]: Did you include this by accident? Looks unrelated to the PR.

Also, `:has` only has 91.66% global compatibility: https://caniuse.com/css-has

File: src/sass/table/table.scss [hunk 3]
```diff
@@ -64,6 +68,34 @@ $density-relaxed-img-radius: 8px;
     padding: $density-relaxed-padding;
 }
 
+.table--frozen-header thead {
+    position: sticky;
```
Comment (agliga) [inline src/sass/table/table.scss:72]: Does this solution also work with dropdowns inside the cell?

File: src/sass/table/table.scss (hunk 2 above)
Comment (ArtBlue) [reply src/sass/table/table.scss:168]: It's not an accident. This is to highlight the row on selection. Are you pulling down PRs locally when you review?

This doesn't impact function and is purely decorative. We do use `:has` for similar types of progressive enhancements, for which current level of support is not really an issue.

File: src/sass/table/table.scss (hunk 1 above)
Comment (ArtBlue) [reply src/sass/table/table.scss:71]: We aren't necessarily bound to the vernacular of DS. We chart our own course for what makes sense for the web/users. Often times, DS actually reevaluates their terminology and follows suit as well.

I used `frozen` instead of `sticky` since that is par for the course on user-level understanding of table cells. We can rename if needed, but I'm not seeing the naming as an issue.

File: src/sass/table/table.scss (hunk 3 above)
Comment (ArtBlue) [reply src/sass/table/table.scss:72]: Good catch. Let me implement the full markup inside with open dropdowns and check. If it doesn't, I'll check for a `css` solution, but we may end up having to do something via `js` in ebay ui.

File: src/sass/table/table.scss (hunk 2 above)
Comment (ArtBlue) [reply src/sass/table/table.scss:168]: The markup was already done for selection. This is just a styling change; not all of those are specified granularly in PRs. No worries. Better safe than sorry. 😄

File: src/sass/table/table.scss (hunk 3 above)
Comment (ArtBlue) [reply src/sass/table/table.scss:72]: @agliga , it works fine, at least with listbox. I have a feeling the issues the other team was having had to do with incorrect `z-index` and lack of `background-color` on `<thead>` and all other non-data columns (outer wrapper).

<kbd>image</kbd>

**Review by agliga — APPROVED** [review]
LGTM!

**Comment by ArtBlue** [comment]
> LGTM, just the freeze-column count feels a bit weird to me. Unless we have usage that ppl are using skin standalone and building bare foot table without wrapping it with component, I think it would be more flexible just add class to `tr > td,th` and ebayui/table will expose column count then apply to each `tr > td,th`.

Our strategy is typically to lean on static implementation of features without requiring major work from external libraries, and the smallest footprint to accomplish the goal. This makes it more optimal.

1. It allows the `css` module to work anywhere. Since we are an open-source library, we always want to lean on that.
2. It is far less brittle. If a user forgets to add the the marker to any `td` or `th`, the table will break.
3. It is more efficient as fewer changes/updates need to be made on fewer elements.

BTW, the column freezing is not a column count like you see on typical spreadsheets. It freezes only a specific column to avoid the table being unusable in smaller screens.
